### PR TITLE
CORE-9190 Add input/output ticket fields to Job JSON

### PR DESCRIPTION
**Pull request by psarando**
This PR will add a `Ticket` field to the `StepInput` JSON, and an `OutputDirTicket` field to the `Job` JSON.

This PR will also add `InputTicketsFile` and `OutputTicketFile` fields to the `Job` JSON, and a `FilterInputsWithTickets` method to collect all inputs with a ticket, for use in `condor-launcher`.

### CODE REVIEW

File: jobs.go [hunk 1]
```diff
@@ -76,16 +76,19 @@ type Job struct {
...
 	IRODSBase          string         `json:"irods_base"`
 	Name               string         `json:"name"`
 	NFSBase            string         `json:"nfs_base"`
 	Notify             bool           `json:"notify"`
 	NowDate            string         `json:"now_date"`
-	OutputDir          string         `json:"output_dir"`   //the value parsed out of the JSON. Use OutputDirectory() instead.
-	RequestDisk        string         `json:"request_disk"` //untested for now
+	OutputDir          string         `json:"output_dir"`         //the value parsed out of the JSON. Use OutputDirectory() instead.
+	OutputDirTicket    string         `json:"output_dir_ticket"`  //the write ticket for output_dir (assumes output_dir is set correctly).
+	OutputTicketFile   string         `json:"output_ticket_list"` //path to the file of the output dest with ticket (not from upstream).
```
Comment (ianmcorvidae) [inline jobs.go]: What are these `File` ones and what does "not from upstream" mean?

Comment (ianmcorvidae) [reply jobs.go]: And why are their JSON keys `_list`, I guess?

Comment (psarando) [reply jobs.go]: These will be populated by `condor-launcher`, so they don't come from the JSON submitted by the `apps` service (I was being lazy and copied this terminology from other comments in this Job JSON).

`condor-launcher` will create these "Path List" files, but only if input/output tickets exist, then populate the fields with a path to the files generated on the submit node. Then if these `_list` fields are not blank, the paths to these files will be added to the `transfer_input_files` list in `iplant.cmd`.

Comment (ianmcorvidae) [reply jobs.go]: Ah, cool, makes sense. Thanks for the explanation!

**Comment by psarando** [comment]
I'll go ahead and merge+tag this now for `jex-adapter` and `condor-launcher`, but reviews are still welcome.

**Pull request merged**